blueocean-dashboard/src/main/js/components/Pipelines.jsx

```diff
@@ -14,24 +15,47 @@ const translate = i18nTranslator('blueocean-dashboard');
 
 @observer
 export class Pipelines extends Component {
-    componentWillMount() {
-        this._initPager(this.props);
+    onChange = debounce(e => {
+        this.context.router.push(`${this.props.location.pathname}${this.updateGetParam('search', e)}`);
+    }, 200);
+
+    getSearchText() {
+        return this.props.location.query.search ? decodeURIComponent(this.props.location.query.search) : '';
```

Comment:
You can either do a null check here, or you acn pass in a location prop into the component in the test such that...

```javascript
{
    location: {
        query: {}
    }
}
```